tensorflow_federated/python/learning/keras_utils_test.py

```diff
@@ -451,6 +451,108 @@ def test_keras_model_and_optimizer(self):
+        keras_model=keras_model,
+        dummy_batch=dummy_batch,
+        loss=[tf.keras.losses.MeanSquaredError(),
+              tf.keras.losses.MeanSquaredError(),
+              tf.keras.losses.MeanSquaredError()])
+
+      self.assertIsInstance(tff_model, model_utils.EnhancedModel)
+      output = tff_model.forward_pass(dummy_batch)
+      self.assertAllClose(output.loss, 2.0)
+
+    with self.subTest("loss_dict_no_opt"):
+      tff_model = keras_utils.from_keras_model(
+        keras_model=keras_model,
+        dummy_batch=dummy_batch,
+        loss={'dense': tf.keras.losses.MeanSquaredError(),
+              'dense_1': tf.keras.losses.MeanSquaredError(),
+              'dense_2': tf.keras.losses.MeanSquaredError()})
+
+      self.assertIsInstance(tff_model, model_utils.EnhancedModel)
+      output = tff_model.forward_pass(dummy_batch)
+      self.assertAllClose(output.loss, 2.0)
+
+    with self.subTest("trainable_model"):
+      tff_model = keras_utils.from_keras_model(
+        keras_model=keras_model,
+        dummy_batch=dummy_batch,
+        loss=[tf.keras.losses.MeanSquaredError(),
+              tf.keras.losses.MeanSquaredError(),
+              tf.keras.losses.MeanSquaredError()],
+        optimizer=tf.keras.optimizers.SGD(learning_rate=0.01))
+
+      self.assertIsInstance(tff_model, model_utils.EnhancedTrainableModel)
+      self.assertTrue(hasattr(tff_model._model._keras_model, 'optimizer'))
+      output = tff_model.forward_pass(dummy_batch)
+      self.assertAllClose(output.loss, 2.0)
+
+    keras_model = model_examples.build_multiple_outputs_keras_model()
+    with self.subTest("loss_weights_as_list"):
+      tff_model = keras_utils.from_keras_model(
+        keras_model=keras_model,
+        dummy_batch=dummy_batch,
+        loss=[tf.keras.losses.MeanSquaredError(),
+              tf.keras.losses.MeanSquaredError(),
+              tf.keras.losses.MeanSquaredError()],
+        loss_weights=[0.1, 0.2, 0.3])
+
+      output = tff_model.forward_pass(dummy_batch)
+      self.assertAllClose(output.loss, 0.5)
+
+    with self.subTest("loss_weights_as_dict"):
+      tff_model = keras_utils.from_keras_model(
+        keras_model=keras_model,
+        dummy_batch=dummy_batch,
+        loss=[tf.keras.losses.MeanSquaredError(),
+              tf.keras.losses.MeanSquaredError(),
+              tf.keras.losses.MeanSquaredError()],
+        loss_weights={'dense_5': 0.1, 'dense_6': 0.2, 'dense_7': 0.3})
+
+      output = tff_model.forward_pass(dummy_batch)
+      self.assertAllClose(output.loss, 0.5)
+
+    with self.subTest("loss_weights_assert_fail"):
+      with self.assertRaises(ValueError):
+        _ = keras_utils.from_keras_model(
+          keras_model=keras_model,
+          dummy_batch=dummy_batch,
+          loss=[tf.keras.losses.MeanSquaredError(),
+                tf.keras.losses.MeanSquaredError(),
+                tf.keras.losses.MeanSquaredError()],
+          loss_weights=[0.1, 0.2])
+
```

Comment:
Lets add one more test with a dict of loss_weights that doesn't have the same keys are the outputs.